examples/noise-js/index.js

```diff
@@ -0,0 +1,320 @@
+import { createLightNode } from "js-waku/lib/create_waku";
+import { utils } from "js-waku";
+import { waitForRemotePeer } from "js-waku/lib/wait_for_remote_peer";
+import {
+  Fleet,
+  getPredefinedBootstrapNodes,
+} from "js-waku/lib/predefined_bootstrap_nodes";
+import { PeerDiscoveryStaticPeers } from "js-waku/lib/peer_discovery_static_list";
+import { Protocols } from "js-waku";
+import * as noise from "@waku/noise";
+import protobuf from "protobufjs";
+import QRCode from "qrcode";
+// TODO: Get rid of these
+import hexToArrayBuffer from "hex-to-array-buffer";
+import arrayBufferToHex from "array-buffer-to-hex";
+
+const messagesDiv = document.getElementById("messages");
+const nicknameInput = document.getElementById("nick-input");
+const textInput = document.getElementById("text-input");
+const sendButton = document.getElementById("send-btn");
+const sendingStatusSpan = document.getElementById("sending-status");
+const chatArea = document.getElementById("chat-area");
+const qrCanvas = document.getElementById("qr-canvas");
+const qrUrl = document.getElementById("qr-url");
+const wakuStatusSpan = document.getElementById("waku-status");
+const handshakeStatusSpan = document.getElementById("handshake-status");
+
+function getPairingInfofromUrl() {
+  const urlParts = window.location.href.split("?");
+  if (urlParts.length < 2) return undefined;
+
+  const pairingParts = decodeURIComponent(urlParts[1]).split(":");
+  if (pairingParts.length < 6)
+    throw new Error("invalid pairing information format");
+
+  const qrMessageNameTag = new Uint8Array(
+    hexToArrayBuffer(pairingParts.shift())
+  );
+
+  return new noise.InitiatorParameters(
+    pairingParts.join(":"),
```

Comment:
This is a bit strange, it looks like we are doing `string hex -> Array Buffer -> string`. I am guessing `pairingParts.join` uses `toString`?

I see that `InitiatorParameters` takes a string called `qrCode`.

I'd rather have it take the data in the most native format (Uint8Array?) and let the implementation decide how to transfer the data (e.g. a QR code does not have to be used by the application).

Reply:
Because data comes from URL we have to encode/decode messageNameTag to and from Buffer so I do not see a way around it. 
What I did is I moved to a new format of URL where it follows this template `host/?messageNameTag&qrCodeString`.
Now it is a lot less hustle for decoding/encoding and only necessary minimus is kept: hex <-> buffer, uri encode/decode.